### PR TITLE
Fix tarball file

### PR DESCRIPTION
**Pull request by arcanis**
**Summary**

The offline mirror feature accidentally disabled the codepath that was used for installing packages from `file:` protocol. The tests didn't caught this error because they're using a wrapper class, `LocalTarballFetcher`, that wasn't affected by this issue.

### CODE REVIEW

**Comment by arcanis** [comment]
Wait, the revert shouldn't be there, I'll strip it

**Comment by arcanis** [comment]
👍

**Comment by bestander** [comment]
Looks like a few tests failed

**Comment by arcanis** [comment]
Should be fine now

File: __tests__/commands/install/lockfiles.js [hunk 1]
```diff
@@ -10,11 +10,16 @@ import {getPackageVersion, runInstall} from '../_helpers.js';
 import {promisify} from '../../../src/util/promise';
 
 jasmine.DEFAULT_TIMEOUT_INTERVAL = 150000;
-
+test.concurrent = test.skip;
```
Comment (bestander) [inline __tests__/commands/install/lockfiles.js]: oops :)

**Comment by bestander** [comment]
@arcanis, you forgot to reenable tests

File: __tests__/commands/install/lockfiles.js (hunk 1 above)
Comment (arcanis) [inline __tests__/commands/install/lockfiles.js]: :scream:

**Comment by bestander** [comment]
This PR fixes relative paths but not absolute ones both on Windows and Unix.
Is it safe to add a root path to the regex, @arcanis?

**Comment by arcanis** [comment]
I could change the regex to: `/^(?!\.\.\/|\/|[a-z]:[\\\/])/`, what do you think?

**Comment by arcanis** [comment]
It seems to catch everything: https://regex101.com/r/zS5bmW/1

**Comment by bestander** [comment]
Looks good to me, can you send a PR when you have a chance?

**Pull request merged**